runtime/cudaq/algorithms/sample.h

```diff
@@ -173,6 +173,16 @@ auto runSamplingAsync(KernelFunctor &&wrappedKernel, quantum_platform &platform,
 }
 } // namespace details
 
+/// \brief Sample options to provide to the sample() / async_sample() functions
+///
+/// \param shots number of shots to run for the given kernel (default is to use
+/// value stored in platform, or 1000).
+/// \param noise noise model to use for the sample operation
+struct sample_options {
+  std::size_t shots = cudaq::get_platform().get_shots().value_or(1000);
+  cudaq::noise_model noise;
+};
```

Comment:
Style question: do we prefer as written of something more like:
```suggestion
struct sample_options {
  /// number of shots to run for the given kernel (default is to use
  /// value stored in platform, or 1000).
  std::size_t shots = cudaq::get_platform().get_shots().value_or(1000);

  /// noise model to use for the sample operation
  cudaq::noise_model noise;
};
```

I guess I'm more used to the edited style, which involves fewer special doxygen tags, etc.

Reply:
Those result in different descriptions appearing in VSCode. Here is what it shows up as right now:
![Screenshot 2023-09-19 145243](https://github.com/NVIDIA/cuda-quantum/assets/141149032/b19f74ea-6aec-474e-b528-edc8e6dcd11e)

And if I make the suggested change, it only shows up like this:
![Screenshot 2023-09-19 145200](https://github.com/NVIDIA/cuda-quantum/assets/141149032/1c5e7577-1875-49f5-b1b4-7dd842f10573)

So I think I prefer it as it is right now?